app/queries/find_by_source_metadata_identifier.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+class FindBySourceMetadataIdentifier
+  def self.queries
+    [:find_by_source_metadata_identifier, :find_by_source_metadata_identifiers]
+  end
+
+  attr_reader :query_service
+  def initialize(query_service:)
+    @query_service = query_service
+  end
+
+  # Find resources by their source metadata identifier. Has logic to handle that
+  # identifier potentially being an Alma ID on input.
+  def find_by_source_metadata_identifier(source_metadata_identifier:)
+    result = query_service.custom_queries.find_by_property(property: :source_metadata_identifier, value: source_metadata_identifier)
+    # If given a BibID which is for Alma, see if there's a non-Alma input.
+    if result.blank? && RemoteRecord.alma?(source_metadata_identifier)
+      non_alma_id = source_metadata_identifier.match(/^99([\d]*)3506421/)&.[](1)
+      return result if non_alma_id.nil?
+      return query_service.custom_queries.find_by_property(property: :source_metadata_identifier, value: non_alma_id)
+    end
+    result
+  end
+
+  def find_by_source_metadata_identifiers(source_metadata_identifiers:)
+    alma_ids = source_metadata_identifiers.select do |bib_id|
```

Comment:
Same question as previous. Should we just run both versions for every ID in case it's a voyager ID that fits the alma pattern?

Reply:
Ah, what a shame. I can do that I guess, but it'll increase the number of queries by a fair number and I have some concern about the number of boolean queries. We can do it by ear though, and it'd only effect performance of thumbnail resolution for the catalog search page I think.